# Laboratory excercise 5

The notebook contains excercise connected to the auditory excerice 6. For any questions feel free to contact assistant: eda.jovicic@fer.hr

In [9]:
import pandas as pd
import numpy as np
import statsmodels.formula.api as smf

np.random.seed(2)

1. Load the cleaned dataset from the first exercise. If you haven't saved the dataset, rerun the exercise and save the final dataset.

In [10]:
file_path = r'C:\Users\Kristina Vrdoljak\Desktop\AppCogSci\data science\Week 3\2nd homework assignment-20240104\X_clean.csv'
df = pd.read_csv(file_path)
df

Unnamed: 0  Gender  EthnicGroup  ParentEduc     LunchType  TestPrep  \
0               0  female            2           1      standard         1   
1               1  female            2           5      standard         1   
2               2  female            1           3      standard         1   
3               3    male            0           0  free/reduced         1   
4               4    male            2           5      standard         1   
...           ...     ...          ...         ...           ...       ...   
29112       29112  female            3           2      standard         1   
29113       29113    male            4           2      standard         1   
29114       29114  female            2           2  free/reduced         0   
29115       29115  female            3           0      standard         0   
29116       29116    male            1           5      standard         1   

       ParentMaritalStatus  PracticeSport  IsFirstChild  NrSiblings  \
0                        1              1            10         3.0   
1                        1              2            10         0.0   
2                        2              2            10         4.0   
3                        1              0             3         1.0   
4                        1              2            10         0.0   
...                    ...            ...           ...         ...   
29112                    2              2             3         2.0   
29113                    2              1             3         1.0   
29114                    1              2             3         1.0   
29115                    1              1             3         3.0   
29116                    1              0             3         1.0   

       TransportMeans  WklyStudyHours  MathScore  ReadingScore  WritingScore  \
0                   1               1         71            71            74   
1                   2               0         69            90            88   
2                   1               1         87            93            91   
3                   2               0         45            56            42   
4                   1               0         76            78            75   
...               ...             ...        ...           ...           ...   
29112               1               0         59            61            65   
29113               0               0         58            53            51   
29114               0               0         61            70            67   
29115               1               0         82            90            93   
29116               1               0         64            60            58   

       Gender_male  LunchType_standard  
0                0                   1  
1                0                   1  
2                0                   1  
3                1                   0  
4                1                   1  
...            ...                 ...  
29112            0                   1  
29113            1                   1  
29114            0                   0  
29115            0                   1  
29116            1                   1  

[29117 rows x 17 columns]

2. Using linear regression, model the MathScore using features: TestPrep, NrSiblings, ParentMaritalStatus and ReadingScore. 

In [4]:
mod=smf.ols(formula='MathScore~C(TestPrep)+ C(ParentMaritalStatus) + NrSiblings + ReadingScore',data=df)
res=mod.fit()
res.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              MathScore   R-squared:                       0.670
Model:                            OLS   Adj. R-squared:                  0.669
Method:                 Least Squares   F-statistic:                     9829.
Date:                Tue, 09 Jan 2024   Prob (F-statistic):               0.00
Time:                        19:15:51   Log-Likelihood:            -1.0471e+05
No. Observations:               29117   AIC:                         2.094e+05
Df Residuals:                   29110   BIC:                         2.095e+05
Df Model:                           6                                         
Covariance Type:            nonrobust                                         
=====================================================================================================
                                        coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------------------------------
Intercept                             6.2302      0.314     19.867      0.000       5.616       6.845
C(TestPrep)[T.none]                   1.1060      0.113      9.807      0.000       0.885       1.327
C(ParentMaritalStatus)[T.married]     0.1733      0.145      1.193      0.233      -0.112       0.458
C(ParentMaritalStatus)[T.single]     -0.1318      0.168     -0.786      0.432      -0.460       0.197
C(ParentMaritalStatus)[T.widowed]     0.5432      0.396      1.373      0.170      -0.232       1.319
NrSiblings                            0.0155      0.036      0.428      0.668      -0.055       0.086
ReadingScore                          0.8570      0.004    239.510      0.000       0.850       0.864
==============================================================================
Omnibus:                      233.486   Durbin-Watson:                   1.987
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              155.618
Skew:                          -0.010   Prob(JB):                     1.61e-34
Kurtosis:                       2.642   Cond. No.                         562.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

Questions:

a) Using the summary, write down the formula with coefficients

$$MathScore =  6.2302 \times C(TestPrep)[0] + 7.3362 \times C(TestPrep)[1] + 0.1733 \times C(ParentMaritalStatus)[T.1] + (- 0.1318) \times C(ParentMaritalStatus)[T.2] + 0.5432 \times C(ParentMaritalStatus)[T.3] + 0.0155 \times NrSiblings + 0.8570 \times ReadingScore$$

b) Are there any statisticaly important features?

$TestPrep (p < 0,005) and ReadingScore (p < 0,005).$

c) Let's interpret the data, if students do test preparation, how does that affect the outcome on the math exam? 

If students do test preparation, MathScore will be bigger for $6.2302$.

d) Are students who practice sport better at math exams or worse? Using linear regression, model the MathScore using only one feature: PracticeSport. Don't include intercept. Using summary, decide which students do the best at math exams - those who never practice sport, sometimes practice sport or those who do it regularly?

In [5]:
mod=smf.ols(formula='MathScore~C(PracticeSport)-1', data=df)
res=mod.fit()
res.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              MathScore   R-squared:                       0.006
Model:                            OLS   Adj. R-squared:                  0.006
Method:                 Least Squares   F-statistic:                     87.35
Date:                Tue, 09 Jan 2024   Prob (F-statistic):           1.51e-38
Time:                        19:16:13   Log-Likelihood:            -1.2074e+05
No. Observations:               29117   AIC:                         2.415e+05
Df Residuals:                   29114   BIC:                         2.415e+05
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
===============================================================================================
                                  coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------------------------
C(PracticeSport)[never]        64.1576      0.244    262.426      0.000      63.678      64.637
C(PracticeSport)[regularly]    67.8441      0.150    452.998      0.000      67.551      68.138
C(PracticeSport)[sometimes]    66.2896      0.126    526.520      0.000      66.043      66.536
==============================================================================
Omnibus:                      234.462   Durbin-Watson:                   1.997
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              214.722
Skew:                          -0.171   Prob(JB):                     2.36e-47
Kurtosis:                       2.756   Cond. No.                         1.94
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

Students that do the best at math exams are those who practice sport regulary because the coefficient is the largest for $$C(PracticeSport)[1]=67.8441.$$

Type Markdown and LaTeX:  𝛼2